### ISSUE TITLE
Proposal to add a feature/widget to make widgets editable and shows the updates while typing.

### ISSUE THREAD

**Issue by williamzheng0711** [flutter/flutter#86465]
## Use case

It is nowadays common to use a live text editor to process words stuff. For example, the app "Notion" and "Tencent Document（騰訊文檔）" gives such functionality. This means as long as you know the basic Markdown syntax, you can see what you typed immediately, don't need to see another "preview" area of your text. 

Now there are packages that convert plain text to Markdown widgets by analyzing the text's content and do the parsing. But those packages do not provide an **editable** thing, i.e., you always need to go somewhere to preview and then switch back to continue your editing. 

**Hence a Notion-like LIVE text editor is still in lack.** 

## Proposal

Actually, I think any widget with the following 2 things will do: 1) textEditingController, which makes widget editable, and 2) child, which enables us to show the (converted) thing we are typing.

**Comment by darshankawar** [comment]
@WilliamZheng99 
There's a related open request to add `Rich Text` based editor to the framework. Check it out and see if it resembles your request. https://github.com/flutter/flutter/issues/12675

**Comment by WilliamZheng99** [comment]
@darshankawar I read that post before (last night) and I also tried using Zefyr and Quill but both have some errors and what' more, they cannot provide a Live text editing effect. 
I am thinking that if the only way is to write a new widget to achieve that.

**Comment by darshankawar** [comment]
@WilliamZheng99 
Have you looked at https://pub.dev/packages/flutter_markdown

**Comment by WilliamZheng99** [comment]
@darshankawar Yes. But that is not editable. I am looking at it for its parsing detail rather

**Comment by HansMuller** [comment]
CC @justinmc @Renzo-Olivares

**Comment by justinmc** [comment]
I like this proposal.  I'm not sure whether or not Flutter should include a widget that does this out of the box, but as we move forward with more advanced text editing features, we should definitely at least make sure it's possible to build something like this.

**Comment by MonolithFoundation** [comment]
Does this RFC still considers? Or, maybe ad support in flutter_markdown package? such as an editable widget with markdown preview?
